app/scripts/services/stats.js

```diff
@@ -168,11 +173,27 @@ angular.module('stats', ['sql', 'health', 'tableinfo'])
       }).error(onErrorResponse);
     };
 
+    var refreshShardInfo = function () {
+      if (!data.online) return;
+
+      ShardInfo.executeTableStmt()
```

Comment:
we said we don't want to execute the same shard query twice, because it is slow.
the shard information must be re-used for both the table view and node view.